### PR TITLE
Docs: Add UPDATE describtion for spark3.

### PR DESCRIPTION
**Pull request by coolderli**
I found the doc's lack of `UPDATE` queries on spark3. And I added it.

### CODE REVIEW

**Comment by coolderli** [comment]
cc @rdblue

File: site/docs/spark-writes.md [hunk 1]
```diff
@@ -171,6 +172,18 @@ WHERE ts >= '2020-05-01 00:00:00' and ts < '2020-06-01 00:00:00'
 
 If the delte filter matches entire partitions of the table, Iceberg will perform a metadata-only delete. If the filter matches individual rows of a table, then Iceberg will rewrite only the affected data files.
 
+### `UPDATE`
+
+Spark 3 added support for `UPDATE` queries to remove data from tables.
```
Comment (rdblue) [inline site/docs/spark-writes.md:177]: I think this description isn't quite right. It should be
> Spark 3.1 added support for `UPDATE` queries that update matching rows in tables.

It would also be good to add a pointer to `MERGE INTO` for more complex updates.

**Comment by rdblue** [comment]
Thanks, @coolderli! Good to have this added to the docs.

File: site/docs/spark-writes.md (hunk 1 above)
Comment (coolderli) [reply site/docs/spark-writes.md]: I've fixed it. I'll think about the complex updates for `MERGE INTO`  and add in another patch.

Comment (rdblue) [reply site/docs/spark-writes.md]: Let's do it here while we're editing the `UPDATE` section.

Comment (coolderli) [reply site/docs/spark-writes.md]: @rdblue I have added some updates for `MERGE INTO`, do I have any other omissions?

File: site/docs/spark-writes.md [hunk 2]
```diff
@@ -89,6 +90,16 @@ Inserts also support additional conditions:
...
+Source rows can also come from the target table or a subset of it.
+
+```sql
+MERGE INTO prod.db.target t
+USING (SELECT * from prod.db.target where op = 'increment') s
+ON t.id = s.id
+WHEN MATCHED AND t.count <> 0 THEN UPDATE SET t.count = 0, t.op = 'delete'
+WHEN MATCHED AND t.count = 0 THEN UPDATE SET t.count = s.count + 1
+```
+
```
Comment (rdblue) [inline site/docs/spark-writes.md:102]: This change looks unrelated and is probably not a good example of how you would merge a table into itself because the `op` field would probably not be stored in the target. I think it's a good idea to show that you can merge a table into itself, but that's probably a different PR where we can discuss what we want in the example. I would definitely include a recommendation to add partition filters.

File: site/docs/spark-writes.md (hunk 1 above)
Comment (rdblue) [reply site/docs/spark-writes.md]: I think you misunderstood what I was suggesting. Instead of changing `MERGE INTO`, I think that the `UPDATE` docs should point the reader to `MERGE INTO` for more complex cases. Something like this with a link:

> For more complex row-level updates based on incoming data, see the section on `MERGE INTO`.

Comment (coolderli) [reply site/docs/spark-writes.md]: Sorry about that. I have updated.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks for reminding me, @coolderli! I'll merge this.
